lesson16/index.js

```diff
@@ -0,0 +1,124 @@
+// const url = "https://myjson.dit.upm.es/api/bins/hyv9";
+const url = "https://myjson.dit.upm.es/api/bins/d9vx";
+const newsBlock = document.getElementById("news");
+const newsTabs = document.getElementById("news_Tabs");
+
+const createNewsTabs = (allData) => {
+  allData.forEach((content, index) => {
+    const li = document.createElement("li");
+    li.classList.add("tab_item");
+    li.id = content.category;
+    li.textContent = content.category;
+    newsTabs.appendChild(li);
+    if(content.isActive === true){
+      li.classList.add("is-active");
+    }
+  });
+};
+
+const createNewsContent = (allData) => {
+  // 初期表示用の処理
+  const primaryElement = document.createElement("div");
+  primaryElement.classList.add("news_Wrapper");
+  const lists = document.createElement("ul");
+  lists.classList.add("news_Lists");
+  lists.id = 'news_Lists';
+  const primaryImageWrapper = document.createElement("div");
+  primaryElement.appendChild(lists);
+  primaryElement.appendChild(primaryImageWrapper);
+  newsBlock.appendChild(primaryElement);
+
+  const selectedData = allData.find((data) => data.isActive === true);
+  primaryImageWrapper.classList.add("news_Img");
+  primaryImageWrapper.id = 'news_Img';
+  const img = document.createElement("img");
+  img.src = selectedData.image.img;
+  primaryImageWrapper.appendChild(img);
+
+  renderNewsList(selectedData);
+};
+
+const setTabClickEvent = (allData) => {
+  // タブクリック時の処理
+  newsTabs.addEventListener("click", (event) => {
+    // event.targetにis-activeをつける
+    // event.targetのidを取得し、それに紐づく記事と画像を取得
+    const target = event.target;
+
+    // タブのindexを取得し、スタイル変更
+    const activeTab = newsTabs.querySelector(".is-active");
+    activeTab.classList.remove("is-active");
+    target.classList.add("is-active");
+
+    const targetData = allData.find((d) => d.category === target.id);
+    setCategoryImage(targetData);
+    renderNewsList(targetData);
+  });
+}
+
+
+const setCategoryImage = (targetData) => {
+  const imgWrapper = document.getElementById("news_Img");
+  imgWrapper.innerHTML = "";
+  const img = document.createElement("img");
+  img.src = targetData.image.img;
+  img.alt = targetData.image.alt;
+  imgWrapper.appendChild(img);
+}
+
+const renderNewsList = (article) => {
+  const newsList = document.getElementById("news_Lists");
+  newsList.innerHTML = "";
+  Object.keys(article.article).forEach((key) => {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    const comment = document.createElement("span");
+    a.href = article.article[key].url;
+    a.textContent = article.article[key].title;
+    const dateDiff = getDateDiff(article.article[key].publishedDate);
+    if (dateDiff < 14) {
```

Comment:
Q.
Why is the number 14 here?

Reply:
I had set a larger number so that I wouldn't have to rewrite the JSON data.
There is no particular meaning other than that.
(I really thought 3 days would be better, but I did this for convenience.)

Reply:
なるほど...。
レビュワーとして、この点は仕様と違うので気になりました。
また、この後もう一人からレビューをいただくと思いますが、その際にも同じコメントを受ける可能性があります。
面倒かもしれませんが、レビュー中はJSONデータを適宜修正し3日以内であればNEWアイコンを表示するようにする方がいいと思いました。